server/broadcast.go

```diff
@@ -544,6 +548,40 @@ func transcodeSegment(cxn *rtmpConnection, seg *stream.HLSSegment, name string,
 		}
 	}
 
+	// [EXPERIMENTAL] send content detection results to callback webhook
+	if DetectionWebhookURL != "" && len(res.Detections) > 0 {
+		go func(mid core.ManifestID, seqNo uint64, detections []*net.DetectData) {
+			type DetectionWebhookRequest struct {
+				ManifestID          core.ManifestID               `json:"manifestID"`
+				SeqNo               uint64                        `json:"seqNo"`
+				SceneClassification []net.SceneClassificationData `json:"sceneClassification"`
+			}
+			req := DetectionWebhookRequest{ManifestID: mid, SeqNo: seqNo}
+			for _, detection := range detections {
+				switch x := detection.Value.(type) {
+				case *net.DetectData_SceneClassification:
+					data := x.SceneClassification
+					req.SceneClassification = append(req.SceneClassification, *data)
+				}
+			}
+			jsonValue, err := json.Marshal(req)
+			if err != nil {
+				glog.Errorf("Unable to marshal detection result into JSON manifestID=%v seqNo=%v", mid, seqNo)
+				return
+			}
+			resp, err := http.Post(DetectionWebhookURL, "application/json", bytes.NewBuffer(jsonValue))
```

Comment:
The standard package level HTTP client doesn't specify a timeout. We should probably create a HTTP client that at the very least specifies a timeout to avoid a hanging detection webhook server hijacking a bunch of goroutines.